src/dsymbol.c

```diff
@@ -470,6 +507,39 @@ Dsymbol *Dsymbol::searchX(Loc loc, Scope *sc, RootObject *id)
             sm = s->search(loc, (Identifier *)id);
             break;
 
+        case DYNCAST_TYPE:
+        {
+            Type *index = (Type*)id;
+            Expression *expr = NULL;
+            Type *t = NULL;
+            Dsymbol *sym = NULL;
+
+            index->resolve(loc, sc, &expr, &t, &sym);
+            if (expr)
+            {
+                sm = takeTypeTupleIndex(loc, sc, s, id, expr);
+            }
+            else if (t)
+            {
+                index->error(loc, "expected an expression as index, got a type (%s)", t->toChars());
+                return NULL;
+            }
+            else
+            {
+                index->error(loc, "index is not an expression");
+                return NULL;
+            }
+            break;
+        }
+
+        case DYNCAST_EXPRESSION:
+            sm = takeTypeTupleIndex(loc, sc, s, id, (Expression*)id);
```

Comment:
For all casts, insert a space before `*`.
s/`(Expression*)id`/`(Expression *)id`/